src/snowflake/snowpark/functions.py

```diff
@@ -1309,7 +1309,20 @@ def approx_percentile(
         <BLANKLINE>
     """
     c = _to_col_if_str(col, "approx_percentile")
-    return builtin("approx_percentile", _emit_ast=_emit_ast)(c, lit(percentile))
+    # Build AST here to prevent `percentile` from being recorded as a literal instead of float.
+    if _emit_ast:
+        ast = proto.Expr()
+        build_builtin_fn_apply(
+            ast,
+            "approx_percentile",
+            *tuple(
+                snowpark_expression_to_ast(arg) if isinstance(arg, Expression) else arg
+                for arg in [c, percentile]
+            ),
+        )
```

Comment:
This body of code repeats identically across all standalone functions. Its 3 parameters are `_emit_ast`, the function name (e.g. `"approx_percentile"`), and the list that maps to the argument tuple (e.g. `[c, percentile]`). I think it's worth refactoring this into a separate helper, it will reduce repetitiveness a lot.